storage: resolve intents synchronously for splits/merges

tbg opened the pull request:
The resolve allowance isn't known to ever have fired for splits/merges
(it might be trigger-able with giant start/end keys), but it worried me
when I ran across it in:

https://github.com/cockroachdb/cockroach/issues/28005#issuecomment-408632803

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/31538)
<!-- Reviewable:end -->

bdarnell reviewed (approved):
> Reviewed 1 of 1 files at r1.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/31538)* status: :shipit: complete! 0 of 0 LGTMs obtained



<!-- Sent from Reviewable.io -->

tbg committed:
storage: resolve intents synchronously for splits/merges

The resolve allowance isn't known to ever have fired for splits/merges
(it might be trigger-able with giant start/end keys), but it worried me
when I ran across it in:

https://github.com/cockroachdb/cockroach/issues/28005#issuecomment-408632803

Release note: None

tschottdorf commented:
bors r=bdarnell
